### ISSUE TITLE
Add TensorBoard logging and metrics

### ISSUE THREAD

**Issue by CooperBigFoot** [CooperBigFoot/RL-demo#8]
## Description
Implement comprehensive logging and visualization for training progress using TensorBoard.

## Reference
See planning document: `/scratchpads/phase3_integration_plan.md` - Section 3

## Dependencies
- #7 (Training script must be working)

## Tasks
- [ ] Create `src/rl_demo/utils/logger.py`
- [ ] Implement TensorBoard logger wrapper:
  - Episode rewards (mean, min, max)
  - Training loss
  - Epsilon value
  - Episode length
  - Constraint violations (floods, droughts)
- [ ] Add reservoir-specific metrics:
  - Average storage level
  - Total hydropower generated
  - Environmental flow violations
- [ ] Create visualization helpers:
  - Plot episode trajectory
  - Storage level over time
  - Action distribution
- [ ] Update training script to use logger

## Code Structure
```python
from torch.utils.tensorboard import SummaryWriter

class ReservoirLogger:
    def __init__(self, log_dir="runs/dqn_reservoir"):
        self.writer = SummaryWriter(log_dir)
        self.episode_count = 0
        
    def log_episode(self, rewards, storage_levels, actions):
        self.writer.add_scalar("episode/total_reward", sum(rewards), self.episode_count)
        self.writer.add_scalar("episode/avg_storage", np.mean(storage_levels), self.episode_count)
        # Add histograms, etc.
```

## Acceptance Criteria
- TensorBoard shows all key metrics
- Can monitor training in real-time
- Metrics help diagnose training issues
- Visualization is clear and informative

## Priority
Medium - Important for debugging and monitoring
